pkg/networkservice/chains/endpoint/server.go

```diff
@@ -55,12 +56,12 @@ type endpoint struct {
 //             - name - name of the NetworkServiceServer
 //             - tokenGenerator - token.GeneratorFunc - generates tokens for use in Path
 //             - additionalFunctionality - any additional NetworkServiceServer chain elements to be included in the chain
-func NewServer(name string, authzPolicy *rego.PreparedEvalQuery, tokenGenerator token.GeneratorFunc, additionalFunctionality ...networkservice.NetworkServiceServer) Endpoint {
+func NewServer(name string, authzPolicies []opa.AuthorizationPolicy, tokenGenerator token.GeneratorFunc, additionalFunctionality ...networkservice.NetworkServiceServer) Endpoint {
```

Comment:
Could we simply make this an authzPolicy 'NetworkServiceServer' rather than a list of opa policies?

So instead of passing authzPolicies, I'd pass authorize.NewServer(authzPolicies...) 

Makes the system nice and general :)